extension.js

```diff
@@ -154,7 +155,30 @@ var GESTURE_DOWN = 1;
 var GESTURE_LEFT = 2;
 var GESTURE_RIGHT = 3;
 
-function gesture(kind) {
+function swap_for_natural(gesture) {
+    switch(gesture) {
+        case GESTURE_UP:
+            return GESTURE_DOWN
+        case GESTURE_DOWN:
+            return GESTURE_UP
+        case GESTURE_RIGHT:
+            return GESTURE_LEFT
+        case GESTURE_LEFT:
+            return GESTURE_RIGHT
+        default:
+            return gesture
+    }
+}
+
+function apply_scroll_settings(gesture) {
+    if (touchpad_settings && touchpad_settings.get_boolean('natural-scroll')) {
+        return swap_for_natural(gesture)
+    }
+    return gesture
+}
+
+function gesture(direction) {
+    const kind = apply_scroll_settings(direction)
     if (kind === GESTURE_UP) {
```

Comment:
Again, not an issue with your PR, but @jpttrssn would you be wiling to change this to a `switch`, just for constancy and neatness.

Reply:
Absolutely! I would actually prefer to create functions for each gesture and call them directly from Touchegg avoiding the conditionals and the need for the gesture() function. I think doing this may also remove the need for the GESTURE_* constants. What do you think?

Reply:
Go for it!